code/modules/projectiles/gun.dm

```diff
@@ -1635,7 +1677,7 @@ not all weapons use normal magazines etc. load_into_chamber() itself is designed
 
 	// Full auto fucks your scatter up big time
 	// Note that full auto uses burst scatter multipliers
-	if(flags_gun_features & GUN_FULL_AUTO_ON)
+	if(gun_firemode == GUN_FIREMODE_AUTOMATIC)
```

Comment:
This makes me think that capturing the firemodes in a datum would be for the best. We can then would have much more flexibility in how things worked